_posts/2022-10-25-using-h3-hexes.md

```diff
@@ -0,0 +1,190 @@
+
+### H3 Uses Math, Not Spatial Operations
+{: .text-left}
+
+To understand how H3 solves these problems, we need to understand how it works. Coming from the GIS world, we think of the "point in polygon" problem spatially: search to see if the given point is inside the given area. We can then aggregate or summarize all the data within that area to come up with a single number or metric. However, with H3 we need to shift our thinking. **H3 takes a lat/long and assigns an ID that corresponds to a mathematically-defined hex.** There's nothing inherently spatial about it, it's all just math. The output is just a number.
+
+### It's a Common Grid (kind of)
+{: .text-left}
+
+As an open-source system, H3 has become a sort of de facto standard. Anyone can use it, and anyone can take someone else's data with H3 grid ids and compare it with their own. While it's been around for a few years, there could be a novelty factor driving it's adoption, so it remains to be seen how widely it will be adopted.
+
+However, even if it doesn't become the overwhelming global standard, the H3 system can still be useful for **standardizing analyses across projects within an organization.** If we at UGRC do a project for one state agency with the grid, it would be easy to pull that data into another project in the future knowing that they both use the same hexes.
+
+### H3 Handles Different Grid Resolutions Elegantly
+{: .text-left}
+
+It is mathematically impossible[^1] to create a larger hexagon from smaller hexagons, which means we can't create a _geographical_ hierarchy of hexes where one level of hexes perfectly cover the same area as a single larger hex. However, H3 makes a _logical_ hierarchy by rotating the larger, parent hex to cover as much of its seven children as possible, as visible in the image below. The H3 [indexing page](https://h3geo.org/docs/highlights/indexing) has more detail on how this works.
+
+![H3 Hexagons at various scales]({% link images/h3_hexes.png %})
+{: .flex .flex--center}
+
+The important point is that an individual hex ID contains the IDs of the hexes "above" it in the hierarchy, kind of like how census GeoIDs contain the block, tract, county, and state codes. This makes it easy to traverse up to a larger hex without re-calculating the IDs.
+
+Using these multiple scales (which H3 calls resolutions), we can **create hexes at multiple scales to aggregate our data to larger and larger areas.** With these, we can view trends at the neighborhood, city, or county level. We can also create web maps that allow us to drill down from general trends to neighborhood or block level to see what areas contribute the most.
+
+### Assigning H3 IDs is Fast
+{: .text-left}
+
+As noted above, H3 uses some fancy mathematics under the hood to assign hex ids instead of doing spatial analysis. Written in C and exposed through a variety of bindings in other languages, these calculations are **pretty darn fast at figuring out what hex cell a point belongs in.**
```

Comment:
**question**: is this math done on the client side or does it use an api that could disappear?

Reply:
I'm 95% sure it's all local. I haven't found an explicit statement that is the case, but there's a ton of C code in the source. My C is rusty (no pun intended) but I'm pretty sure the algorithm implementation is in there.